File: 01/jaeyoung/53_Maximum_Subarray.cpp
class Solution {
   public:
    int maxSubArray(vector<int>& nums) {
        const int INF = 1000000000;
        int max_sum = -INF, cur_sum = -INF;

        for (int num : nums) {
            cur_sum = max(num, cur_sum + num);
            max_sum = max(max_sum, cur_sum);
        }

        return max_sum;
    }
};

File: 01/hyeongseok/53.Maximum_Subarray.ts
function maxSubArray(nums: number[]): number {
    let maxSum = Number.MIN_SAFE_INTEGER
    let curSum = 0;
    for(let i = 0; i < nums.length; i++) {
        curSum += nums[i];
        maxSum = Math.max(curSum, maxSum);
        if (curSum < 0) {
            curSum = 0;
        }
    }
    return maxSum;
};